Adds CONTRIBUTING file

Gerlison opened the pull request:
### Descrição das Mudanças ###

<!--
Primeiramente muito obrigado por sua contribuição, antes de abrir o PR tenha certeza de preencher todos os campos necessários, o não preenchimento dos campos pode resultar em um atraso para revisão do seu código ou até mesmo fechamento do PR.
-->

Adicionado o arquivo CONTRIBUTING.md

### *Bugs* Corrigidos ###

<!-- 
Se for mais de uma issue, liste seguindo o padrão 
-->

[#1 ]


### Mudanças na API ###

Nenhuma.

### Mudanças de comportamento ###

Nenhuma.

### Testing Procedure ###

Nenhum.

### PR Checklist ###
<!-- A ser preenchido por você -->

- [ ] Possui testes (se não, por favor descreva o motivo de não ter teste)
- [X] Foi feito Rebase para a master no momento que o PR foi aberto
- [X] Modificações estão de acordo com o padrão de código
- [X] Atualizado documentação

**Não possui teste devido ser adição de documentação.**

### PR Checklist  do time ###
<!-- A ser preenchido por quem for revisar o código -->

- [X] O PR está apontando para a branch correta
- [ ] Os testes estão passando (ou as falha não são relacionadas a este PR)

Commit:
Adds CONTRIBUTING file

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+# Contribuindo
+
+Uma vez que o _Idvogados_ é um projeto totalmente **open source**, qualquer pessoa/instituição pode contribuir da forma que desejar. Para isso basta seguir o guia de contribuição do projeto e nosso [código de conduta]().
```
Nós já temos o códig de conduta

```suggestion
Uma vez que o _Idvogados_ é um projeto totalmente **open source**, qualquer pessoa/instituição pode contribuir da forma que desejar. Para isso basta seguir o guia de contribuição do projeto e nosso [código de conduta](https://github.com/idvogados/codigo-de-conduta/blob/master/README.md).
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+## Linguagem
+
+- _Commits_ devem ser feitos em inglês.
+- _Issues_, _pull requests_ e revisões podem ser feitas em português para facilitar a comunicação e chegar a um consenso rapidamente.
```
Acho que é melhor utilizar a palavra 'devem' ao invés de 'podem'. 

```suggestion
- _Issues_, _pull requests_ e revisões devem ser feitas em português para facilitar a comunicação e chegar a um consenso rapidamente.
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+## Desenvolvimento
+
+- Desenvolva apenas em seu _branch_.
+- Faça _rebase_ com o **develop** **antes** de criar _pull requests_.
+- Faça _rebase_ do **develop** em seu _branch_ diariamente.
```
Eu e o @pictos estamos utilizando a branch dev no backend. Seria interessante manter padronizado para todos os projetos.

O que acha de deixar dev?
```suggestion
- Faça _rebase_ com o **dev** **antes** de criar _pull requests_.
- Faça _rebase_ do **dev** em seu _branch_ diariamente.
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+3. Na branch criada, faça suas alterações.
+4. Realize o _commit_ seguindo este guia.
+5. Abra um _pull request_ mencionando a _issue_ relacionada.
+6. Aguarde as aprovações e faça o _merge_ na **develop**
```
Mesmo questionamento acima.
```suggestion
6. Aguarde as aprovações e faça o _merge_ na **dev**
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+#### Como começar
+
+1. Crie um _fork_ do projeto.
+2. No seu _fork_, crie uma branch a partir da develop. (seguimos o [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow))
```
Mesmo questionamento acima.
```suggestion
2. No seu _fork_, crie uma branch a partir da dev. (seguimos o [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow))
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+#### O que **não** fazer
+
+- Desenvolver direto no **develop**.
+- _Merge_ de _branches_ no **develop** antes que o código seja revisado.
+- _Merge_ de _branches_ no **develop** antes que todas as discussões tenham se encerrado.
```
Mesmo questionamento acima.
```suggestion
- Desenvolver direto no **dev**
- _Merge_ de _branches_ no **dev** antes que o código seja revisado.
- _Merge_ de _branches_ no **dev** antes que todas as discussões tenham se encerrado.
```

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+  - [_How to Write a Git Commit Message_](https://chris.beams.io/posts/git-commit/)
+- Seja breve e suscinto no título das mensagens do _commit_.
+- Use o corpo da mensagem para explicações adicionais.
+- Se esforce para manter a **develop** o mais limpo possível.
```
Mesmo questionamento a cima.
```suggestion
- Se esforce para manter a **dev** o mais limpo possível.
```

rodrigondec reviewed (changes requested):
Muito obrigado por sua contribuição. Segue algumas alterações, sugestões e questionamentos.

Sinta livre para discordar e discutir :rocket:

Gerlison committed:
Update CONTRIBUTING.md

Add code of conduct's link;
Change branch 'develop' to 'dev';
Change some words;

Co-Authored-By: Rodrigo Castro <rodrigondec@gmail.com>

rodrigondec commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+- Use o corpo da mensagem para explicações adicionais.
+- Se esforce para manter a **dev** o mais limpo possível.
+
+#### O que **não** fazer
```
Existem duas seções "O que não fazer". Seria melhor deixar elas juntas ou mudar o título se for adequado.

Gerlison replied:
Que tal alterar para :
```suggestion
#### O que **não** fazer quando commitando
```

e na outra ocorrencia:
"O que **não** fazer quando contribuindo"

rodrigondec replied:
Agora eu prestei atenção que eles são subheaders em níveis diferentes. 

Mas acho que fica bom deixando específico e explícito com esse "quando [...]".

Commit:
update CONTRIBUNTING.md file

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+## Linguagem
+
+- _Commits_ devem ser feitos em inglês.
```
Ainda não podemos afirmar isso, pois ainda não temos o time de líderes técnicos.

Gerlison replied:
Sim, eu ja supus isso, mas temos que partir de alguma premissa no momento. Assim que definido apenas a lingua alteramos aqui. O que acha?

pictos replied:
Pode ser, mas como não será a versão final irei alterar o target pra `develop`.

Gerlison replied:
O @rodrigondec sugeriu seguirmos o adotado no backend como **dev**. O que acha?

pictos replied:
isso mesmo, digitei o nome longo xD `e errado xP`

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+- Desenvolva apenas em seu _branch_.
+- Faça _rebase_ com o **dev** **antes** de criar _pull requests_.
+- Faça _rebase_ do **dev** em seu _branch_ diariamente.
```
Não acho que seja necessário fazer rebase todos os dias. Acredito que a pessoa garantir que tem as atualizações mais recentes da *dev* antes de começar a trabalhar já é o bastante.

Gerlison replied:
O 'diariamente' seria o caso do cenário ideal pra que não haja conflitos. Muitas das observações tirei de alguns exemplos que encontrei.  Acha que deveria remover essa linha?

pictos replied:
Acho que sim. Ou talvez trocar por `Mantenha seu fork sempre atualizado`. 
https://help.github.com/pt/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

rodrigondec replied:
@pictos sobre essa questão, atualmente é uma das restrições nas branchs. Se for o caso se não termos essa restrição me avisem da decisão para eu remover da proteção das branchs.

![image](https://user-images.githubusercontent.com/3301060/80547061-a16eab00-898d-11ea-9206-0d1ef7462bbc.png)

pictos replied:
@rodrigondec, isso podemos ver no futuro. Por enquanto acho que pode deixar as configurações assim.

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+#### Como começar
+
+1. Crie um _fork_ do projeto.
+2. No seu _fork_, crie uma branch a partir da dev. (seguimos o [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow))
```
Ainda não podemos afirmar isso, pois ainda não temos o time de líderes técnicos.

Gerlison replied:
Novamente, estou partindo da premissa que optaremos por seguir o padrão do GitFlow, que inclusive ja foi mencionado na reuniao. Mas caso tenha alterações com os tech leads, basta atualizar aqui.

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+3. Na branch criada, faça suas alterações.
+4. Realize o _commit_ seguindo este guia.
+5. Abra um _pull request_ mencionando a _issue_ relacionada.
+6. Aguarde as aprovações e faça o _merge_ na **dev**
```
Isso dá ideia de que o autor do PR é responsável pelo *merge*, o que é não é correto. A responsabilidade do merge é de algum líder técnico ou alguém dentro do projeto que tenha permissão para isso.

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+5. Abra um _pull request_ mencionando a _issue_ relacionada.
+6. Aguarde as aprovações e faça o _merge_ na **dev**
+
+> Como criar um [fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo)?
+> Como criar um [pull request de um fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)?
```
Amei essas referências ❣️

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+#### O que **não** fazer quando contribuindo
+
+- Desenvolver direto no **dev**
+- _Merge_ de _branches_ no **dev** antes que o código seja revisado.
```
A pessoa não terá permissão para isso. Então não precisamos desse item.

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+## Commits
+
+- Faça _commits_ pequenos, que descrevem alterações únicas.
+- Siga as referências sobre como escrever mensagens de _commits_:
```
Acho que aqui podemos ser um pouco menos duro, pois não sabemos se vamos seguir este padrão. 
```suggestion
- Sugerimos que siga as recomendações sobre como escrever mensagens de _commits_ descritas neste link:
```

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+  - [_How to Write a Git Commit Message_](https://chris.beams.io/posts/git-commit/)
+- Seja breve e suscinto no título das mensagens do _commit_.
+- Use o corpo da mensagem para explicações adicionais.
+- Se esforce para manter a **dev** o mais limpo possível.
```
Isso não é tarefa da pessoa que vai contribuir e sim dos líderes técnicos. Então acredito que isso pode ser removido

rodrigondec replied:
Talvez mover essas coisas para um arquivo interno nosso de 'instruções' para os tech leads. 

Poderíamos deixar esses arquivos expostos no repositório por exemplo. Facilita se tiver transição de tech leads (os arquivos que ele precisa ler e prestar atenção já são públicos).

E traria mais transparência do processo de gerência para quem esta contribuindo (ter uma gerência mística nem sempre é bom).

O que acham?

Gerlison replied:
Acho uma boa. Você poderia abrir a issue para tal?

pictos replied:
Acredito que uma nova issue seria uma boa.

rodrigondec replied:
Já 'temos' uma issue para isso. A #2 discute sobre o que é um PR válido (algo de responsabilidade dos tec leads).

Vou colocar na descrição da issue essa sugestão. Acho bom esperarmos os tech leads antes de tomar essa decisão. Ou então envolver outras pessoas (@pabrrs, @vilchesalves, @pictos) na decisão.

pictos commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+- Descreva qual é o problema em questão.
+- Se possível, dê soluções para o problema.
+- Faça _links_ para código ou para referências a outras _issues_/_pull requests_ se necessário.
+- Todas as contribuições devem ser baseadas em uma _issue_ aberta.
```
Uma pequena adição aqui.
```suggestion
- Todas as contribuições devem ser baseadas em uma _issue_ aberta e marcadas como pronta para implementar.
```

pictos reviewed (approved):
@Gerlison muito obg pela contribuição, algumas observações. 🚀

pictos reviewed (changes requested):
apertei errado 🙈

Gerlison committed:
Update CONTRIBUTING.md

Co-Authored-By: Pedro Jesus <pedrojesus.cefet@gmail.com>

pictos reviewed (approved):
Acredito que está tudo ok. @Gerlison obg pelo PR. E um lembrete amigável, parece que seu cliente git está com o force-push ativado. Isso dificulta a fazer o review, pois não há um `diff`, se puder desativá-lo nos próximos PRs seria muito bom. Mais um vez obg <3

Gerlison commented:
> Acredito que está tudo ok. @Gerlison obg pelo PR. E um lembrete amigável, parece que seu cliente git está com o force-push ativado. Isso dificulta a fazer o review, pois não há um `diff`, se puder desativá-lo nos próximos PRs seria muito bom. Mais um vez obg <3

@pictos O force-push foi devido a um squash que fiz em alguns commits automaticos aqui do comentarios. Por terem a mesma mensagem e alterarem apenas um linha ou palavra, juntem em apenas 1 pra facilitar a review. Por isso o force.
